### ISSUE TITLE
KlarnaPaymentView "Bankkonto" won't render in android

### ISSUE THREAD

**Issue by zpengyu** [klarna/react-native-klarna-inapp-sdk#41]
**Describe the bug**
This issue occurred in android, after session initialised, the app rendered three payment options:
1) Bankkonto 
2) Köp först. Betala sen 
3) Dela upp

When the user click "1) Bankkonto", KlarnaPaymentView#onLoaded was triggered, but the webview won't render. 2) and 3) work as expected only if when the user hasn't clicked 1) previously.

**To Reproduce**
Steps to reproduce the behavior (scenario 1):
1. Render 3 klarna payment options
2. choose to render "bankkonto"
3. webview of bankkonto won't render

Steps to reproduce the behavior (scenario 2):
1. Render 3 klarna payment options
2. choose to render "köp först betala sen", then choose "bankkonto"
3. webview of bankkonto won't render

**Logs from android studio**
client url and credentials are masked by "xxxxxxxxxxxxxxxxxxx".
```
I/chromium: [INFO:CONSOLE(0)] "Uncaught (in promise) TypeError: Failed to fetch", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (0)
    [INFO:CONSOLE(11)] "Received message: 
    {
      "action": "actionToComponent",
      "messageId": "7302ce5",
      "params": {
        "actionType": "initialize",
        "clientToken": "xxxxxxxxxxxxxx_client_token_xxxxxxxxxxxxxx",
        "returnUrl": "https://www.xxxxxxxxxxxxxxxxxxx.se/klarna-success-return"
      },
      "receiver": "KlarnaPaymentsWrapper",
      "sender": "Native"
    }", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (11)
    [INFO:CONSOLE(11)] "Sending event: RecvdInitialize", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (11)
    [INFO:CONSOLE(11)] "Handling mock init.", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (11)
I/chromium: [INFO:CONSOLE(11)] "Token parsed:  true", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (11)
I/chromium: [INFO:CONSOLE(11)] "Sending event: ValidatedInitialize", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (11)
I/chromium: [INFO:CONSOLE(11)] "Performing Klarna.Payments.init().", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (11)
    [INFO:CONSOLE(11)] "Sending event: PerformingInitialize", source: file:///data/user/0/xxxxxxxxxxxxxxxxxxxxxx/files/kp_index.html?mockkp=true&storeall=true&loglevel=0 (11)
W/WebView: java.lang.Throwable: A WebView method was called on thread 'JavaBridge'. All WebView methods must be called on the same thread. (Expected Looper Looper (main, tid 2) {5db60d0} called on Looper (JavaBridge, tid 1446) {39cb602}, FYI main Looper is Looper (main, tid 2) {5db60d0})
        at android.webkit.WebView.checkThread(WebView.java:2575)
        at android.webkit.WebView.getUrl(WebView.java:1251)
        at com.klarna.mobile.sdk.a.e.m.g.<init>(AnalyticBuilder.kt:2)
        at com.klarna.mobile.sdk.a.e.m.b.a(AnalyticBuilder.kt:1)
        at com.klarna.mobile.sdk.a.e.m.a.a(AnalyticBuilder.kt:4)
        at com.klarna.mobile.sdk.a.e.m.a.a(AnalyticBuilder.kt:12)
        at com.klarna.mobile.sdk.core.webview.g$a.postMessage(WebViewNativeHook.kt:36)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:336)
        at android.os.Looper.loop(Looper.java:174)
        at android.os.HandlerThread.run(HandlerThread.java:67)
D/StrictMode: StrictMode policy violation: android.os.strictmode.WebViewMethodCalledOnWrongThreadViolation
        at android.webkit.WebView.checkThread(WebView.java:2575)
        at android.webkit.WebView.getUrl(WebView.java:1251)
        at com.klarna.mobile.sdk.a.e.m.g.<init>(AnalyticBuilder.kt:2)
        at com.klarna.mobile.sdk.a.e.m.b.a(AnalyticBuilder.kt:1)
        at com.klarna.mobile.sdk.a.e.m.a.a(AnalyticBuilder.kt:4)
        at com.klarna.mobile.sdk.a.e.m.a.a(AnalyticBuilder.kt:12)
        at com.klarna.mobile.sdk.core.webview.g$a.postMessage(WebViewNativeHook.kt:36)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:336)
        at android.os.Looper.loop(Looper.java:174)
        at android.os.HandlerThread.run(HandlerThread.java:67)
W/System.err: java.lang.RuntimeException: java.lang.Throwable: A WebView method was called on thread 'JavaBridge'. All WebView methods must be called on the same thread. (Expected Looper Looper (main, tid 2) {5db60d0} called on Looper (JavaBridge, tid 1446) {39cb602}, FYI main Looper is Looper (main, tid 2) {5db60d0})
        at android.webkit.WebView.checkThread(WebView.java:2580)
        at android.webkit.WebView.getUrl(WebView.java:1251)
        at com.klarna.mobile.sdk.a.e.m.g.<init>(AnalyticBuilder.kt:2)
        at com.klarna.mobile.sdk.a.e.m.b.a(AnalyticBuilder.kt:1)
        at com.klarna.mobile.sdk.a.e.m.a.a(AnalyticBuilder.kt:4)
        at com.klarna.mobile.sdk.a.e.m.a.a(AnalyticBuilder.kt:12)
        at com.klarna.mobile.sdk.core.webview.g$a.postMessage(WebViewNativeHook.kt:36)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:336)
        at android.os.Looper.loop(Looper.java:174)
        at android.os.HandlerThread.run(HandlerThread.java:67)
    Caused by: java.lang.Throwable: A WebView method was called on thread 'JavaBridge'. All WebView methods must be called on the same thread. (Expected Looper Looper (main, tid 2) {5db60d0} called on Looper (JavaBridge, tid 1446) {39cb602}, FYI main Looper is Looper (main, tid 2) {5db60d0})
        at android.webkit.WebView.checkThread(WebView.java:2575)
    	... 10 more
```

**Device and version:**
 - Device: google pixel 4
 - OS: android 10
 - react-native-klarna-inapp-sdk: 2.0.5 (currently latest)

**Additional context**
In 2.0.5, webview won't render, but in 2.0.2, it will crash the app;
It works perfectly in iOS.

**Comment by omarbh** [comment]
Hi @zpengyu, we will look into this and get back to you as soon as possible.

**Comment by NMGuner** [comment]
Hi @zpengyu, I am looking into this issue and I can see the same logs you posted when running. Those are internal issues which will be looked upon on a future release. As for the issue of not rendering a WebView or a crash I was unable to recreate the problem, I tried `2.0.2` and `2.0.5` versions as you mentioned. Can you provide us the stacktrace of those two cases? I may be able to recreate or understand the problem from those logs.

**Comment by zpengyu** [comment]
Hi @NMGuner Thanks for the reply! 

Here're some updates about the issue we have:
Both 2.0.2 and 2.0.5 output the same log, the only difference is 2.0.5 won't crash the app, I think after upgrading KlarnaMobileSDK to 2.0.15, the app won't crash.

The stacktrace is basically the same I pasted in my previous comment, and it complained `java.lang.Throwable: A WebView method was called on thread 'JavaBridge'. All WebView methods must be called on the same thread.`

It seems like the `GuardedRunnable` is deprecated in `PaymentViewWrapper.java` could this be the case that cause the problem?

**Comment by NMGuner** [comment]
@zpengyu Thanks for the information.

The logs you see are actually coming from the native `In-App SDK` hence the `GuardedRunnable` in the plugin implementation is not the cause of the problem. Those logs are apparently happening due to logging implementation inside our native SDK and will be looked upon. 

On my tries I observed that the logs from native sdk does not crash the react native app(they are caught) and the render on WebView did not have any problems, so I tried to manually trigger something similar from a new `Thread` that I created and that crashed the app with an `Exception`. I do think that such exception from logging errors wouldn't cause WebView not to render especially if caught but it might crash the app. 

As a solution we will look into the issue inside the `Klarna In-App SDK` and provide a fix for the `Thread` used on such logs to prevent any errors or logs that might occur. After that fix is released it will be carried over to the React Native SDK as well.

**Comment by pontus1** [comment]
As the KlarnaPaymentView triggers requestLayout() the value from "document.body.scrollHeight" is sometimes "null" (thus returning) or "0" setting scaledHeight to 0

**Comment by zpengyu** [comment]
We semi-solved this issue from our side by patching the file PaymentViewWrapper.java

The issue occasionally occurred during rendering the web view of pay via bank account. This web view wouldn't get rendered because the height value cannot be set correctly. Usually the height got updated twice, the first one is 0, the second one is the real value. However, the height value of bankkonto can be updated only once which is always 0, due to the rendering process has been halted for unknown reasons, probably multi threading issue in react native or some illegal characters inside the web content. We can only reach this far and hard to dig deeper.

To solve (or semi-solve) the issue, we gave the height a default value, for example 500, instead of zero from the web view. So the web view will always displayed even though it's not fully rendered. This has been tested many times, one out of three the web view of bank account won't rendered correctly, but at least we can display content what we have.

```
PaymentViewWrapper.java    L78
if(value == null || value.equals("null") || value.equals("undefined")){
    return;
}
+ if(value.equals("0")) {
+     value = "500";
+ }
```
The screenshot below is an example of web view not being fully rendered. 

<img width="358" alt="Screenshot 2020-06-16 at 13 40 22" src="https://user-images.githubusercontent.com/1212952/84769961-054d4180-afd7-11ea-9d98-6fa61fedab4d.png">

So imho, one direction is to investigate the web content of bankkonto. For example, checking some analytics code, strange characters or error handling promises.
